mobile/app/forgot-password.tsx

```diff
@@ -3,34 +3,50 @@ import { Screen } from "../components/Screen";
 import { useTranslation } from "react-i18next";
 import { Icon } from "../components/Icon";
 import { router } from "expo-router";
-import { YStack } from "tamagui";
+import { XStack, YStack } from "tamagui";
 import { Typography } from "../components/Typography";
 import { useForm, Controller } from "react-hook-form";
 import FormInput from "../components/FormInputs/FormInput";
 import Card from "../components/Card";
 import Button from "../components/Button";
 import { useSafeAreaInsets } from "react-native-safe-area-context";
+import { useState } from "react";
+import ChangePasswordConfirmation from "../components/ChangePasswordConfirmation";
+import { ForgotPasswwordPayload, forgotPassword } from "../services/definitions.api";
+import * as Sentry from "@sentry/react-native";
 
 type FormData = {
   email: string;
 };
 
 const ForgotPassword = () => {
+  const insets = useSafeAreaInsets();
+  const { t } = useTranslation("reset");
+  const [emailConfirmation, setEmailConfirmation] = useState(false);
+  const [authError, setAuthError] = useState(false);
+
+  // React Hook form
   const { handleSubmit, control, formState } = useForm<FormData>({
-    defaultValues: { email: "alice@example.com" },
+    defaultValues: { email: "charlie@example.com" },
   });
-
-  const { t } = useTranslation("reset");
   const { errors } = formState;
 
-  const insets = useSafeAreaInsets();
-
-  // TODO: Implement the onSubmit function
-  const onSubmit = (data: FormData) => {
-    console.log("Forgot password for email: ", data.email);
-    router.push("/email-confirmation");
+  // Submit handler - forgot password
+  const onSubmit = async (data: FormData) => {
+    try {
+      const payload: ForgotPasswwordPayload = { email: data.email };
+      await forgotPassword(payload);
```

Comment:
we should use a mutation for this, not call the api function directly, please look up the other examples with useMutation